lms/static/scripts/frontend_apps/components/dashboard/OrganizationActivity.tsx

```diff
@@ -37,23 +33,8 @@ export default function OrganizationActivity() {
 
   useDocumentTitle('All courses');
 
-  const [selectedStudents, setSelectedStudents] = useState<Student[]>([]);
-  const [selectedAssignments, setSelectedAssignments] = useState<Assignment[]>(
-    [],
-  );
-  const [selectedCourses, setSelectedCourses] = useState<Course[]>([]);
-  const studentIds = useMemo(
-    () => selectedStudents.map(s => s.h_userid),
-    [selectedStudents],
-  );
-  const assignmentIds = useMemo(
-    () => selectedAssignments.map(a => `${a.id}`),
-    [selectedAssignments],
-  );
-  const courseIds = useMemo(
-    () => selectedCourses.map(c => `${c.id}`),
-    [selectedCourses],
-  );
+  const { filters, updateFilters } = useDashboardQueryFilters();
+  const { courseIds, assignmentIds, studentIds } = filters;
```

Comment:
Now we can use this hook together with `DashboardActivityFilters`, and the filters will persist in the query string, or we can use "traditional" state management (as in `useState`) and they won't, but `DashboardActivityFilters` doesn't know anything about it and is a fully controlled component.